base/io.jl

```diff
@@ -509,13 +507,17 @@ end
 unsafe_write(s::IO, p::Ptr, n::Integer) = unsafe_write(s, convert(Ptr{UInt8}, p), convert(UInt, n))
 write(s::IO, x::Ref{T}) where {T} = unsafe_write(s, x, Core.sizeof(T))
 write(s::IO, x::Int8) = write(s, reinterpret(UInt8, x))
-function write(s::IO, x::Union{Int16,UInt16,Int32,UInt32,Int64,UInt64,Int128,UInt128,Float16,Float32,Float64})
-    return write(s, Ref(x))
-end
-
 write(s::IO, x::Bool) = write(s, UInt8(x))
 write(to::IO, p::Ptr) = write(to, convert(UInt, p))
 
+function write(io::IO, x::T) where T
+    if isbitstype(T)
+        write(io, Ref(x))
+    else
+        throw(MethodError(write, (io, x)))
```

Comment:
Imo, manually throwing MethodError is almost always confusing. You get a method error but then you see that the method does exist. Why not throw an error that says that only writing bitstype is allowed or something along those lines?

Reply:
Good idea, will do that in the next commit (I will just wait for the padding discussion to resolve).